[2024-06-14T17:05:10.413+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather.weather_data scheduled__2024-06-14T16:45:00+00:00 [queued]>
[2024-06-14T17:05:10.417+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather.weather_data scheduled__2024-06-14T16:45:00+00:00 [queued]>
[2024-06-14T17:05:10.417+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-06-14T17:05:10.423+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): weather_data> on 2024-06-14 16:45:00+00:00
[2024-06-14T17:05:10.425+0000] {standard_task_runner.py:57} INFO - Started process 38 to run task
[2024-06-14T17:05:10.429+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'weather', 'weather_data', 'scheduled__2024-06-14T16:45:00+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/kafka_stream.py', '--cfg-path', '/tmp/tmpusgpgccq']
[2024-06-14T17:05:10.431+0000] {standard_task_runner.py:85} INFO - Job 2: Subtask weather_data
[2024-06-14T17:05:10.474+0000] {task_command.py:415} INFO - Running <TaskInstance: weather.weather_data scheduled__2024-06-14T16:45:00+00:00 [running]> on host e6e8920b48a2
[2024-06-14T17:05:10.534+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather' AIRFLOW_CTX_TASK_ID='weather_data' AIRFLOW_CTX_EXECUTION_DATE='2024-06-14T16:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-14T16:45:00+00:00'
[2024-06-14T17:05:10.557+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.80.8', 29092)]>: connecting to broker:29092 [('192.168.80.8', 29092) IPv4]
[2024-06-14T17:05:10.558+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2024-06-14T17:05:10.558+0000] {conn.py:410} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.80.8', 29092)]>: Connection complete.
[2024-06-14T17:05:10.669+0000] {conn.py:1267} INFO - Broker version identified as 2.5.0
[2024-06-14T17:05:10.670+0000] {conn.py:1268} INFO - Set configuration api_version=(2, 5, 0) to skip auto check_version requests on startup
[2024-06-14T17:05:10.673+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2024-06-14T17:05:10.674+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
[2024-06-14T17:05:10.675+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d169e00 wrapping '2024-06-14' at 0xffff8d1c37b0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'next_ds', '2024-06-14')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.675+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
[2024-06-14T17:05:10.676+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d1a1cc0 wrapping '20240614' at 0xffff8d20ef30 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'next_ds_nodash', '20240614')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.676+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
[2024-06-14T17:05:10.677+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.677+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d1a1200 wrapping '2024-06-14' at 0xffff8d172ab0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_ds', '2024-06-14')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.677+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.677+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d161880 wrapping '20240614' at 0xffff8d172cf0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_ds_nodash', '20240614')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.678+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.678+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
[2024-06-14T17:05:10.679+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.679+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d161b00 wrapping '2024-06-15' at 0xffff8d174970 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'tomorrow_ds', '2024-06-15')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.679+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.680+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d160680 wrapping '20240615' at 0xffff8d1748b0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'tomorrow_ds_nodash', '20240615')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.688+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.689+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d160d40 wrapping '2024-06-13' at 0xffff8d174a30 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'yesterday_ds', '2024-06-13')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.689+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:314 AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.
[2024-06-14T17:05:10.689+0000] {secrets_masker.py:277} WARNING - Unable to redact <Proxy at 0xffff8d160e00 wrapping '20240613' at 0xffff8d1ed370 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'yesterday_ds_nodash', '20240613')>, please report this via <https://github.com/apache/***/issues>. Error was: TypeError: ObjectProxy() missing required argument 'wrapped' (pos 1)
[2024-06-14T17:05:10.671+0000] {kafka_stream.py:62} INFO - {'conf': <airflow.configuration.AirflowConfigParser object at 0xffff925892e0>, 'dag': <DAG: weather>, 'dag_run': <DagRun weather @ 2024-06-14 16:45:00+00:00: scheduled__2024-06-14T16:45:00+00:00, state:running, queued_at: 2024-06-14 17:05:08.149643+00:00. externally triggered: False>, 'data_interval_end': DateTime(2024, 6, 14, 17, 0, 0, tzinfo=Timezone('UTC')), 'data_interval_start': DateTime(2024, 6, 14, 16, 45, 0, tzinfo=Timezone('UTC')), 'ds': '2024-06-14', 'ds_nodash': '20240614', 'execution_date': <Proxy at 0xffff8d169c80 wrapping DateTime(2024, 6, 14, 16, 45, 0, tzinfo=Timezone('UTC')) at 0xffff8d1ba680 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'execution_date', DateTime(2024, 6, 14, 16, 45, 0, tzinfo=Timezone('UTC')))>, 'expanded_ti_count': None, 'inlets': [], 'logical_date': DateTime(2024, 6, 14, 16, 45, 0, tzinfo=Timezone('UTC')), 'macros': <module 'airflow.macros' from '/home/airflow/.local/lib/python3.9/site-packages/airflow/macros/__init__.py'>, 'next_ds': <Proxy at 0xffff8d169e00 wrapping '2024-06-14' at 0xffff8d1c37b0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'next_ds', '2024-06-14')>, 'next_ds_nodash': <Proxy at 0xffff8d1a1cc0 wrapping '20240614' at 0xffff8d20ef30 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'next_ds_nodash', '20240614')>, 'next_execution_date': <Proxy at 0xffff8d1a1c40 wrapping DateTime(2024, 6, 14, 17, 0, 0, tzinfo=Timezone('UTC')) at 0xffff8d1ba6d0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'next_execution_date', DateTime(2024, 6, 14, 17, 0, 0, tzinfo=Timezone('UTC')))>, 'outlets': [], 'params': {}, 'prev_data_interval_start_success': None, 'prev_data_interval_end_success': None, 'prev_ds': <Proxy at 0xffff8d1a1200 wrapping '2024-06-14' at 0xffff8d172ab0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_ds', '2024-06-14')>, 'prev_ds_nodash': <Proxy at 0xffff8d161880 wrapping '20240614' at 0xffff8d172cf0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_ds_nodash', '20240614')>, 'prev_execution_date': <Proxy at 0xffff8d161800 wrapping DateTime(2024, 6, 14, 16, 30, 0, tzinfo=Timezone('UTC')) at 0xffff8d173040 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_execution_date', DateTime(2024, 6, 14, 16, 30, 0, tzinfo=Timezone('UTC')))>, 'prev_execution_date_success': <Proxy at 0xffff8d161840 wrapping None at 0xffff931cd8f0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'prev_execution_date_success', None)>, 'prev_start_date_success': None, 'run_id': 'scheduled__2024-06-14T16:45:00+00:00', 'task': <Task(PythonOperator): weather_data>, 'task_instance': <TaskInstance: weather.weather_data scheduled__2024-06-14T16:45:00+00:00 [running]>, 'task_instance_key_str': 'weather__weather_data__20240614', 'test_mode': False, 'ti': <TaskInstance: weather.weather_data scheduled__2024-06-14T16:45:00+00:00 [running]>, 'tomorrow_ds': <Proxy at 0xffff8d161b00 wrapping '2024-06-15' at 0xffff8d174970 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'tomorrow_ds', '2024-06-15')>, 'tomorrow_ds_nodash': <Proxy at 0xffff8d160680 wrapping '20240615' at 0xffff8d1748b0 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'tomorrow_ds_nodash', '20240615')>, 'triggering_dataset_events': {}, 'ts': '2024-06-14T16:45:00+00:00', 'ts_nodash': '20240614T164500', 'ts_nodash_with_tz': '20240614T164500+0000', 'var': {'json': None, 'value': None}, 'conn': None, 'yesterday_ds': <Proxy at 0xffff8d160d40 wrapping '2024-06-13' at 0xffff8d174a30 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'yesterday_ds', '2024-06-13')>, 'yesterday_ds_nodash': <Proxy at 0xffff8d160e00 wrapping '20240613' at 0xffff8d1ed370 with factory functools.partial(<function lazy_mapping_from_context.<locals>._deprecated_proxy_factory at 0xffff8d181b80>, 'yesterday_ds_nodash', '20240613')>, 'apikey': '***', 'locations': ['Mumbai', 'Pune', 'Surat', 'Kolkata', 'Jammu', 'Bikaner', 'Indore', 'Chennai', 'Bengaluru', 'Ludhiana'], 'days': 2, 'templates_dict': None}
[2024-06-14T17:05:11.846+0000] {conn.py:380} INFO - <BrokerConnection node_id=1 host=broker:29092 <connecting> [IPv4 ('192.168.80.8', 29092)]>: connecting to broker:29092 [('192.168.80.8', 29092) IPv4]
[2024-06-14T17:05:11.847+0000] {conn.py:410} INFO - <BrokerConnection node_id=1 host=broker:29092 <connecting> [IPv4 ('192.168.80.8', 29092)]>: Connection complete.
[2024-06-14T17:05:11.848+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connected> [IPv4 ('192.168.80.8', 29092)]>: Closing connection. 
[2024-06-14T17:05:15.138+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/kafka_stream.py", line 69, in stream_data
    res = get_data(apikey, location, days)
  File "/opt/airflow/dags/kafka_stream.py", line 17, in get_data
    return req.json()
  File "/home/airflow/.local/lib/python3.9/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
[2024-06-14T17:05:15.144+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=weather, task_id=weather_data, execution_date=20240614T164500, start_date=20240614T170510, end_date=20240614T170515
[2024-06-14T17:05:15.151+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 2 for task weather_data (Expecting value: line 1 column 1 (char 0); 38)
[2024-06-14T17:05:15.195+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-06-14T17:05:15.207+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
